frontend/src/main/web/lib/components/glossary/DataTable.jsx

```diff
@@ -0,0 +1,438 @@
+      col: 3,
+      field: 'pos',
+      sort_field: 'part_of_speech'
+    },
+    DESC: {
+      col: 4,
+      field: 'description',
+      sort_field: 'desc'
+    },
+    TRANS_COUNT: {
+      col: 5,
+      field: 'termsCount',
+      sort_field: 'trans_count'
+    }
+  },
+  CELL_HEIGHT: 48,
+
+  propTypes: {
+    glossaryData: React.PropTypes.object.isRequired,
+    glossaryResId: React.PropTypes.arrayOf(
+      React.PropTypes.arrayOf(React.PropTypes.string)
+    ),
+    canAddNewEntry: React.PropTypes.bool.isRequired,
+    canUpdateEntry: React.PropTypes.bool.isRequired,
+    user: React.PropTypes.shape({
+      username: React.PropTypes.string,
+      email: React.PropTypes.string,
+      name: React.PropTypes.string,
+      imageUrl: React.PropTypes.string,
+      languageTeams: React.PropTypes.string
+    }),
+    srcLocale: React.PropTypes.shape({
+      locale: React.PropTypes.shape({
+        localeId: React.PropTypes.string.isRequired,
+        displayName: React.PropTypes.string.isRequired,
+        alias: React.PropTypes.string.isRequired
+      }).isRequired,
+      numberOfTerms: React.PropTypes.number.isRequired
+    }),
+    selectedTransLocale: React.PropTypes.string,
+    totalCount: React.PropTypes.number.isRequired,
+    focusedRow: React.PropTypes.shape({
+      resId: React.PropTypes.string,
+      rowIndex: React.PropTypes.number
+    })
+  },
+
+  getInitialState: function () {
+    var top = 246;
+    return {
+      tbl_width: this._getWidth(),
+      tbl_height: this._getHeight(top),
+      row_height: this.CELL_HEIGHT,
+      header_height: this.CELL_HEIGHT,
+      hoveredRow: -1
+    }
+  },
+
+  _getHeight: function(fixedTop) {
+    var footer = window.document.getElementById("footer");
+    var footerHeight = footer ? footer.clientHeight : 91;
+    var top = _.isUndefined(fixedTop) ? React.findDOMNode(this).offsetTop: fixedTop;
+    var newHeight = window.innerHeight - footerHeight - top;
+
+    //minimum height 250px
+    newHeight = newHeight < 250 ? 250 : newHeight;
+    return newHeight;
+  },
+
+  _getWidth: function () {
+    return window.innerWidth - 48;
+  },
+
+  _handleResize: function(e) {
+    this.setState({tbl_height: this._getHeight(), tbl_width: this._getWidth()});
+  },
+
+  componentDidMount: function() {
+    window.addEventListener('resize', this._handleResize);
+  },
+
+  componentWillUnmount: function() {
+    window.removeEventListener('resize', this._handleResize);
+  },
+
+  _generateTermInfo: function(term) {
+    var title = "";
+    if(!_.isUndefined(term) && !_.isNull(term)) {
+      if (!StringUtils.isEmptyOrNull(term.lastModifiedBy)
+        || !StringUtils.isEmptyOrNull(term.lastModifiedDate)) {
+        title = "Last updated ";
+        if (!StringUtils.isEmptyOrNull(term.lastModifiedBy)) {
+          title += "by: " + term.lastModifiedBy;
+        }
+        if (!StringUtils.isEmptyOrNull(term.lastModifiedDate)) {
+          title += " " + term.lastModifiedDate;
+        }
```

Comment:
Looks like this can print two spaces in a row when the person is not known. I suggest you push each part onto an array, then use `.join(' ')` to get the final string.

```
const parts = ['Last updated']
if (...) {
  parts.push('by:')
  parts.push(term.lastModifiedBy)
}
...
title = parts.join(' ')
```